File: common/src/main/java/dev/obscuria/fragmentum/client/FragmentumClient.java
package dev.obscuria.fragmentum.client;

import dev.obscuria.fragmentum.v2.api.client.ClientTooltipStack;
import dev.obscuria.fragmentum.v2.api.client.TooltipComponentRegistry;
import dev.obscuria.fragmentum.v2.api.common.TooltipStack;
import lombok.experimental.UtilityClass;
import org.jetbrains.annotations.ApiStatus;

@UtilityClass
@ApiStatus.Internal
public final class FragmentumClient {

    public static void init() {
        TooltipComponentRegistry.register(TooltipStack.class, ClientTooltipStack::create);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/FragmentumClient.kt
package dev.obscuria.fragmentum

import dev.obscuria.fragmentum.client.FragmentumClientRegistry
import dev.obscuria.fragmentum.client.AllOfClientTooltipComponent
import dev.obscuria.fragmentum.world.AllOfTooltipComponent

object FragmentumClient {

    fun init() {

        FragmentumClientRegistry.registerTooltipComponent(AllOfTooltipComponent::class.java) {
            AllOfClientTooltipComponent(it)
        }
    }
}